runtime/vm/JVMImage.cpp

```diff
@@ -216,6 +362,48 @@ initializeJVMImage(J9JavaVM *vm)
 	return 0;
 }
 
+extern "C" void
+registerClassLoader(J9ClassLoader *classLoader)
+{
+	JVMImage *jvmImage = JVMImage::getInstance();
+
+	if (jvmImage != NULL) {
```

Comment:
This would cause a silent failure.

If you never expect this case to happen, you can put an assertion instead. (ie `Assert_VM_notNull(jvmImage)`).

If it is possible for it to be NULL then it should return something indicating an error. 

I am going to assume that for this prototype any usage of this function will be guarded by an `IS_COLD_RUN` check, so youre probably okay with an assertion

Reply:
yeah I forgot to put that check in for registering but essentially every call on most of the exposed functions should be guarded by IS_COLD_RUN or IS_WARM_RUN